### PR TITLE
[processor/k8sattributes] Add optional k8s.cluster.uid resource attribute

### PR DESCRIPTION
**Pull request by shivanshuraj1333**
**Description:** 
Add k8s.cluster.uid to attribute to k8sattributes processor and disable it by default for backward compatibility. Users can set it to `true` to populate cluster uid as part of resource attributes.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21974

### CODE REVIEW

**Comment by smithclay** [comment]
Hey, just wanted to ping on this PR. Are you looking for feedback or help? This is some useful context for spans we'd like to see in the k8sattributesprocessor.

File: processor/k8sattributesprocessor/metadata.yaml [hunk 1]
```diff
@@ -8,6 +8,10 @@ status:
 
 # resource attributes are exposed through a different configuration interface (extract::metadata).
 resource_attributes:
+  k8s.cluster.uid:
+    description: Gives cluster uid identified with kube-system namespace
+    type: string
+    enabled: true
```
Comment (dmitryax) [inline processor/k8sattributesprocessor/metadata.yaml:14]: This attribute should be added as optional (disabled by default), otherwise it's a breaking change

Comment (shivanshu1333) [reply processor/k8sattributesprocessor/metadata.yaml:14]: Okay, thanks for reviewing.
Are there any other changes needed apart from this? Would add all the changes.
+ I need to add some tests.

Comment (shivanshu1333) [reply processor/k8sattributesprocessor/metadata.yaml]: Done

**Comment by shivanshu1333** [comment]
@dmitryax can you please take a look?
I've added UTs and e2e and addressed the above review comment.

**Comment by shivanshu1333** [comment]
Hey @kovrus can you please help with reviews, thanks!

File: processor/k8sattributesprocessor/options.go [hunk 2]
```diff
@@ -50,6 +50,9 @@ func withPassthrough() option {
 // enabledAttributes returns the list of resource attributes enabled by default.
 func enabledAttributes() (attributes []string) {
 	defaultConfig := metadata.DefaultResourceAttributesConfig()
+	if defaultConfig.K8sClusterUID.Enabled {
+		attributes = append(attributes, "k8s.cluster.uid")
```
Comment (mwear) [inline processor/k8sattributesprocessor/options.go:54]: There are two references to `k8s.cluster.uid` in this file (here and line 173). Ideally, we would use `conventions.AttributeK8SClusterUID`, but the collector semantic conventions are not up to date. In the meantime can we replace those references with a constant, and put a TODO above the constant to replace it when semconv v.1.21.0 is available?

File: processor/k8sattributesprocessor/internal/kube/client.go [hunk 3]
```diff
@@ -114,7 +114,15 @@ func New(logger *zap.Logger, apiCfg k8sconfig.APIConfig, rules ExtractionRules,
 	}
 
 	if newNamespaceInformer == nil {
-		newNamespaceInformer = newNamespaceSharedInformer
+		// if rules to extract metadata from namespace is configured use namespace shared informer containing
+		// all namespaces including kube-system which contains cluster uid information (kube-system-uid)
+		if c.extractNamespaceLabelsAnnotations() {
+			newNamespaceInformer = newNamespaceSharedInformer
+		} else {
+			// use kube-system shaared informer to only watch kube-system namespace
```
Comment (mwear) [inline processor/k8sattributesprocessor/internal/kube/client.go:122]:
```suggestion
			// use kube-system shared informer to only watch kube-system namespace
```

File: processor/k8sattributesprocessor/options.go (hunk 2 above)
Comment (shivanshu1333) [reply processor/k8sattributesprocessor/options.go]: Yes, was thinking to do it, but then would need to update semconv and bump the deps here.
Thanks for suggested adding a TODO, done!!

File: processor/k8sattributesprocessor/internal/kube/client.go (hunk 3 above)
Comment (shivanshu1333) [reply processor/k8sattributesprocessor/internal/kube/client.go]: Sharp eyes, thanks, done!

File: processor/k8sattributesprocessor/metadata.yaml [hunk 4]
```diff
@@ -5,10 +5,13 @@ status:
   stability:
     beta: [logs, metrics, traces]
   distributions: [contrib, splunk, observiq, sumo, redhat]
-  codeowners:
-    active: [dmitryax, rmfitzpatrick, fatsheep9146]
```
Comment (dmitryax) [inline processor/k8sattributesprocessor/metadata.yaml:9]: Bad rebase?

File: processor/k8sattributesprocessor/testdata/e2e/collector/configmap.yaml [hunk 5]
```diff
@@ -10,6 +10,10 @@ data:
         endpoint: {{ .HostEndpoint }}:4317
         tls:
           insecure: true
+      logging:
+        verbosity: detailed
+        sampling_initial: 5
+        sampling_thereafter: 200
```
Comment (dmitryax) [inline processor/k8sattributesprocessor/testdata/e2e/collector/configmap.yaml:16]: This is an unrelated change

File: processor/k8sattributesprocessor/metadata.yaml (hunk 4 above)
Comment (shivanshu1333) [reply processor/k8sattributesprocessor/metadata.yaml:9]: yes, fixed it

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
@shivanshu1333 please add a changelog entry and fix the linter issues
